### PR TITLE
docs: Add a note about package requirements for fact gathering

### LINKED ISSUE #26148: ansible requires iproute2 package for network facts
##### ISSUE TYPE

 - Bug Report
 - Documentation Report

##### COMPONENT NAME

setup module

##### ANSIBLE VERSION

ansible 2.3.1

##### CONFIGURATION

default

##### OS / ENVIRONMENT

targeet host without iproute2 package

##### SUMMARY

https://github.com/ansible/ansible/blob/v2.3.1.0-1/lib/ansible/module_utils/facts.py#L2460

This information should be avaialbe in facts documentation. And there should be a list of packages which should be installed for each type of fact. Or at least use different source of data for these kind of facts.

##### STEPS TO REPRODUCE

run setup module on host without iproute2 package

##### EXPECTED RESULTS

network interfaces should be available

##### ACTUAL RESULTS

there are no facts about network interfaces.

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Fixes: #26148

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/user_guide/playbooks_vars_facts.rst

### CODE REVIEW

File: docs/docsite/rst/user_guide/playbooks_vars_facts.rst [hunk 1]
```diff
@@ -491,6 +491,19 @@ To reference the system hostname::
 
 You can use facts in conditionals (see :ref:`playbooks_conditionals`) and also in templates. You can also use facts to create dynamic groups of hosts that match particular criteria, see the :ref:`group_by module <group_by_module>` documentation for details.
 
+.. _fact_requirements:
+
+Package requirements for fact gathering
+---------------------------------------
+
+For fact gathering, certain binaries are used. If binary is absent, fact is set to default value.
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_vars_facts.rst]:
```suggestion
Some fact gathering depends on installed binaries on the remote host. If the binary is absent, the fact is set to a default value.
```

File: docs/docsite/rst/user_guide/playbooks_vars_facts.rst [hunk 2]
```diff
@@ -491,6 +491,19 @@ To reference the system hostname::
...
+
+It is recommended to install these binaries on the remote host using default OS pacakge manager.
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_vars_facts.rst]:
```suggestion
We recommend that you install these binaries on the remote host using default OS package manager.
```

File: docs/docsite/rst/user_guide/playbooks_vars_facts.rst [hunk 3]
```diff
@@ -491,6 +491,19 @@ To reference the system hostname::
...
+
+Following is a list of such requirements:
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_vars_facts.rst]:
```suggestion
The following lists required binaries:
```

File: docs/docsite/rst/user_guide/playbooks_vars_facts.rst [hunk 4]
```diff
@@ -491,6 +491,19 @@ To reference the system hostname::
...
+
+* For Linux Network fact gathering, ``ip`` binary is used which comes from ``iproute2`` package.
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_vars_facts.rst]:
```suggestion
* Linux Network fact gathering -  Depends on  the ``ip`` binary from the ``iproute2`` package.
```

File: docs/docsite/rst/user_guide/playbooks_vars_facts.rst [hunk 5]
```diff
@@ -491,6 +491,19 @@ To reference the system hostname::
...
+Package requirements for fact gathering
+---------------------------------------
+
+Some fact gathering depends on installed binaries on the remote host. If the binary is absent, the fact is set to a default value.
+
+We recommended that you install these binaries on the remote host using default OS package manager.
+
+The following lists required binaries:
+
+* Linux Network fact gathering -  Depends on  the ``ip`` binary from the ``iproute2`` package.
```
Comment (bcoca) [inline docs/docsite/rst/user_guide/playbooks_vars_facts.rst]: this might not be correct name for all distros

**Comment by acozine** [comment]
Thanks @Akasurde!

**Pull request merged**